src/pineko/evolve.py

```diff
@@ -284,30 +266,27 @@ def write_operator_card(pineappl_grid, default_card, card_path, tcard):
             "are you sure that's what you want?"
         )
 
-    # For hadronic obs we might need to dump 2 eko cards
-    if conv_type_2 is None or conv_type_1 == conv_type_2:
-        dump_card(card_path, operators_card, conv_type_1)
-    else:
-        # dump card_a
-        dump_card(card_path, operators_card, conv_type_1, suffix=True)
-        # dump card_b
-        dump_card(card_path, operators_card, conv_type_2, suffix=True)
+    # Get the types of convolutions required for this Grid
+    convolutions = pineappl_grid.convolutions
+
+    # TODO: Find a clever way to recombine the same convolutions
```

Comment:
What do you mean ?

Reply:
As it is currently implemented, if we have three convolutions `unpolarized PDF⊗unpolarized PDF⊗ unpolarized FF` then the unpolarized PDF operator will be dumped twice while it should only done **once** and called by the two objects.

Reply:
I see but then the `pineko theory ekos` command will compute the ekos twice ?

Reply:
Okay good then it's fine!

Reply:
After double-checking, this was actually not the case. I pushed a fix in [c792ace](https://github.com/NNPDF/pineko/pull/206/commits/c792ace2dc7ab18a43df50e5ac160a78caa6f65d).

As for the TODO comment, we can simply remove it now.

Reply:
Thanks perfect!